util/conditions/v1beta2/summary.go

```diff
@@ -0,0 +1,134 @@
+// to add a step counter to the generated message by using the WithStepCounter option.
+func NewSummaryCondition(sourceObj runtime.Object, targetConditionType string, opts ...SummaryOption) (*metav1.Condition, error) {
+	summarizeOpt := &SummaryOptions{
+		mergeStrategy: newDefaultMergeStrategy(),
+	}
+	summarizeOpt.ApplyOptions(opts)
+
+	conditions, err := getConditionsWithOwnerInfo(sourceObj)
+	if err != nil {
+		return nil, err
+	}
+
+	expectedConditionTypes := sets.New[string](summarizeOpt.conditionTypes...)
+	ignoreTypesIfMissing := sets.New[string](summarizeOpt.ignoreTypesIfMissing...)
+	existingConditionTypes := sets.New[string]()
+
+	// Drops all the conditions not in scope for the merge operation
+	conditionsInScope := make([]ConditionWithOwnerInfo, 0, len(expectedConditionTypes))
+	for _, condition := range conditions {
+		if !expectedConditionTypes.Has(condition.Type) {
+			continue
+		}
+		conditionsInScope = append(conditionsInScope, condition)
+		existingConditionTypes.Insert(condition.Type)
+	}
+
+	// Add the expected conditions which do net exists, so we are compliant with K8s guidelines
+	// (all missing conditions should be considered unknown).
+
+	diff := expectedConditionTypes.Difference(existingConditionTypes).Difference(ignoreTypesIfMissing).UnsortedList()
+	if len(diff) > 0 {
+		conditionOwner := getConditionOwnerInfo(sourceObj)
+
+		for _, c := range diff {
+			conditionsInScope = append(conditionsInScope, ConditionWithOwnerInfo{
+				OwnerResource: conditionOwner,
+				Condition: metav1.Condition{
+					Type:    c,
+					Status:  metav1.ConditionUnknown,
+					Reason:  NotYetReportedReason,
+					Message: fmt.Sprintf("Condition %s not yet reported", c),
+					// NOTE: LastTransitionTime and ObservedGeneration are not relevant for merge.
+				},
+			})
+		}
+	}
+
+	if len(conditionsInScope) == 0 {
+		return nil, errors.New("summary can't be performed when the list of conditions to be summarized is empty")
+	}
+
+	status, reason, message, err := summarizeOpt.mergeStrategy.Merge(
+		conditionsInScope,
+		summarizeOpt.conditionTypes,
+		sets.New[string](summarizeOpt.negativePolarityConditionTypes...),
+		summarizeOpt.stepCounter,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return &metav1.Condition{
+		Type:    targetConditionType,
+		Status:  status,
+		Reason:  reason,
+		Message: message,
+		// NOTE: LastTransitionTime and ObservedGeneration will be set when this condition is added to an object by calling Set.
+	}, err
+}
+
+// SetSummaryCondition is a convenience method that calls NewSummaryCondition to create a summary condition from the source object,
+// and then calls Set to add the new condition to the target object.
+func SetSummaryCondition(sourceObj, targetObj runtime.Object, targetConditionType string, opts ...MirrorOption) error {
+	mirrorCondition, err := NewMirrorCondition(sourceObj, targetConditionType, opts...)
```

Comment:
```suggestion
	mirrorCondition, err := NewSummaryCondition(sourceObj, targetConditionType, opts...)
```
(btw only noticed because I was clicking into the func and the parameter names didn't match up (sourceConditionType vs targetConditionType :))